app/assets/javascripts/darkswarm/services/cart.js.coffee

```diff
@@ -115,8 +115,9 @@ Darkswarm.factory 'Cart', (CurrentOrder, Variants, $timeout, $http, $modal, $roo
       @line_items = []
       localStorageService.clearAll() # One day this will have to be moar GRANULAR
 
-    has_one_line_item: =>
-      @line_items_finalised.length == 1
+    isOnlyItemInOrder: (id) =>
+      deletedItem = @line_items_finalised.find((item) -> item.id == id)
+      @line_items_finalised.filter((item) -> item.order_id == deletedItem.order_id).length == 1
```

Comment:
please let me know if there is a more elegant way of doing this

Reply:
I'm not entirely sure a Cart can have line items from different orders. If that's the case the problem lays in the Cart abstraction itself. Lines 7 and 8 make me think so. We initialize the line item arrays from a single order :thinking:

Reply:
I think this relates to the option where there's a previous order in the same OC and the OC hasn't closed yet, and the distributor allows editing open orders?

Reply:
that's right @Matt-Yorkley